### PR TITLE
chore: added function to verify duplicates

### PR DESCRIPTION
**Pull request by LeoAdlerr**

### CODE REVIEW

**Comment by LeoAdlerr** [comment]
Please ignore the sonar cloud duplication warning, it is detecting lines such as ['consumo pt vd'] and ['consumo fp vd'] as duplicates

File: src/domain/use-case/ingest-energy-bill.use-case.ts [hunk 1]
```diff
@@ -12,13 +12,14 @@ export class IngestEnergyBill {
...
-  ) { }
+  ) {}
+
   async execute(energyBills: EnergyBillPayload[]) {
     const times: Partial<Time>[] = [];
     const aGroupBills: Partial<EnergyBillGroupA>[] = [];
     const bGroupBills: Partial<EnergyBillGroupB>[] = [];
 
-    console.log(energyBills);
+    console.log('Energy Bills:', energyBills);
```
Comment (lucassbarcelos) [inline src/domain/use-case/ingest-energy-bill.use-case.ts:22]: remove console.log

File: src/domain/use-case/ingest-energy-bill.use-case.ts [hunk 2]
```diff
@@ -94,20 +70,101 @@ export class IngestEnergyBill {
       });
     });
 
-    const savedTimes = await this.timeRepo.save(this.getDistinctObjects(times));
-    console.log(savedTimes);
+    console.log('Times before saving:', times);
```
Comment (lucassbarcelos) [inline src/domain/use-case/ingest-energy-bill.use-case.ts:73]: idem

File: src/domain/use-case/ingest-energy-bill.use-case.ts [hunk 3]
```diff
@@ -94,20 +70,101 @@ export class IngestEnergyBill {
...
+    } catch (error) {
+      console.error('Error saving times:', error);
+    }
+
+    const distinctBGroupBills = this.getDistinctBills(bGroupBills);
+    const distinctAGroupBills = this.getDistinctBills(aGroupBills);
 
-    const savedBGroupBills = await this.billGroupBRepo.save(bGroupBills);
-    console.log(savedBGroupBills);
+    console.log('Distinct B Group Bills:', distinctBGroupBills);
```
Comment (lucassbarcelos) [inline src/domain/use-case/ingest-energy-bill.use-case.ts:94]: idem

File: src/domain/use-case/ingest-energy-bill.use-case.ts [hunk 4]
```diff
@@ -94,20 +70,101 @@ export class IngestEnergyBill {
...
+    console.log('Distinct A Group Bills:', distinctAGroupBills);
```
Comment (lucassbarcelos) [inline src/domain/use-case/ingest-energy-bill.use-case.ts:95]: idem

File: src/domain/use-case/ingest-energy-bill.use-case.ts [hunk 5]
```diff
@@ -94,20 +70,101 @@ export class IngestEnergyBill {
...
+      try {
+        const existsBillb = await this.billGroupBRepo.findOne({
+          where: {
+            month: bBill.month,
+            instalationNumber: bBill.instalationNumber,
+            provider: bBill.provider,
+            plant: bBill.plant
+          },
+        });
+        console.log('Existing B Group Bill:', existsBillb);
```
Comment (lucassbarcelos) [inline src/domain/use-case/ingest-energy-bill.use-case.ts:107]: idem

**Review by lucassbarcelos — APPROVED** [review]

**Pull request merged**